bindings/wasm/examples/lib/evaluate.ts

```diff
@@ -0,0 +1,193 @@
+ */
+export class Evaluator {
+  context: any = {};
+  beforeScript: string = 'resetToCircularDefaults();';
+  afterScript: string =
+      'return typeof result === "undefined" ? undefined : result;';
+
+  protected module?: ManifoldToplevel;
+  protected memoryRegistry: Array<Manifold|CrossSection>;
+
+
+  /**
+   * Construct a new evaluator.
+   *
+   */
+  constructor(module: ManifoldToplevel) {
+    this.module = module;
+    this.memoryRegistry = new Array<Manifold|CrossSection>();
+
+    this.addMembers('Manifold', manifoldMemberFunctions, false);
+    this.addMembers('Manifold', manifoldStaticFunctions, true);
+    this.addMembers('CrossSection', crossSectionMemberFunctions, false);
+    this.addMembers('CrossSection', crossSectionStaticFunctions, true);
+
+    for (const name of toplevelConstructors) {
+      //@ts-ignore
+      const originalFn = module[name];
+      //@ts-ignore
+      this.module[name] = (...args: any) => {
+        const result = originalFn(...args);
+        this.memoryRegistry.push(result);
+        return result;
+      };
+    }
+  }
+
+  /**
+   * Intercept calls and add their results to our garbage collection
+   * list.
+   *
+   * @param className The class to intercept.
+   * @param methodNames An array of methods to intercept.
+   * @param areStatic Are these static methods?  If so, intercept them at
+   * the prototype level.
+   */
+  protected addMembers(
+      className: string, methodNames: Array<string>, areStatic: boolean) {
+    //@ts-ignore
+    const cls = this.module[className];
+    const obj = areStatic ? cls : cls.prototype;
+    for (const name of methodNames) {
+      const originalFn = obj[name];
+      obj[name] = (...args: any) => {
+        //@ts-ignore
+        const result = originalFn(...args);
+        this.memoryRegistry.push(result);
+        return result;
+      };
+    }
+  }
+
+  /**
+   * Delete any objects tagged for garbage collection.
+   */
+  cleanup() {
+    for (const obj of this.memoryRegistry) {
+      // decompose result is an array of manifolds
+      if (obj instanceof Array)
+        for (const elem of obj) elem.delete();
+      else
+        obj.delete();
+    }
+    this.memoryRegistry.length = 0;
+  }
+
+  /**
+   * Evaluate a string as javascript code creating a Manifold model.
+   *
+   * This function assembles the final execution context.  It then runs
+   * `beforeScript`, `code` and `afterScript` in order.  Finally, it
+   * returns the end result.
+   *
+   * @param code The input string.
+   * @returns any By default, this script will return either `undefined`
+   * or a `Manifold` object.  Changing `afterScript` will affect this
+   * behaviour.
+   */
+  evaluate(code: string) {
+    const exposedFunctions = toplevelConstructors.concat(toplevel).map(
+        (name) => [name, (this.module as any)[name]]);
+    const context = {
+      ...Object.fromEntries(exposedFunctions),
+      module: this.module,
+      ...this.context
```

Comment:
Does not automatically include `glMatrix` or `exportModels` anymore; that's up to the caller.

Reply:
The caller may accidentally overwrite context if they're not careful.  I think that follows the principle of least astonishment.